pkg/apis/vpc/v1alpha1/subnet_types.go

```diff
@@ -121,6 +123,10 @@ type SubnetAdvancedConfig struct {
 	EnableVLANExtension bool `json:"enableVLANExtension,omitempty"`
 	// Static IP allocation for VPC Subnet Ports.
 	StaticIPAllocation StaticIPAllocation `json:"staticIPAllocation,omitempty"`
+	// GatewayAddresses specifies custom gateway IP addresses for the Subnet.
+	GatewayAddresses []string `json:"gatewayAddresses,omitempty"`
+	// DHCPServerAddresses specifies custom DHCP server IP addresses for the Subnet.
+	DHCPServerAddresses []string `json:"dhcpServerAddresses,omitempty"`
```

Comment:
It seems in NSX the maximum number of GatewayAddresses/DHCPServerAddresses is one. Maybe we can enforce the same MaxItems.
https://github-vcf.devops.broadcom.net/vcf/nsx/blob/2bc38ea5afd87ae3956af611a0f2e98dfc940a67/mp/policy/policy-api/src/main/resources/api_spec/root/nsx-policy/policy-vpc-subnet.yml#L63-L78

Reply:
good, added